src/neo/SmartContract/ApplicationEngine.Callback.cs

```diff
@@ -0,0 +1,38 @@
+using Neo.SmartContract.Callbacks;
+using Neo.VM.Types;
+using System;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract
+{
+    partial class ApplicationEngine
+    {
+        public static readonly InteropDescriptor System_Callback_Create = Register("System.Callback.Create", nameof(CreateCallback), 0_00000400, TriggerType.All, CallFlags.None);
+        public static readonly InteropDescriptor System_Callback_CreateFromSyscall = Register("System.Callback.CreateFromSyscall", nameof(CreateCallbackFromSyscall), 0_00000400, TriggerType.All, CallFlags.None);
+        public static readonly InteropDescriptor System_Callback_Invoke = Register("System.Callback.Invoke", nameof(InvokeCallback), 0_00000400, TriggerType.All, CallFlags.None);
+
+        internal void InvokeCallback(CallbackBase callback, Array args)
+        {
+            //TODO: We should check the count of the return values
+
+            if (args.Count != callback.ParametersCount)
+                throw new InvalidOperationException();
+            if (callback is PointerCallback pointerCallback)
+                LoadContext(pointerCallback.GetContext());
+            for (int i = args.Count - 1; i >= 0; i--)
+                Push(args[i]);
+            if (callback is SyscallCallback syscallCallback)
+                OnSysCall(syscallCallback.Method);
+        }
+
+        internal PointerCallback CreateCallback(Pointer pointer, int parcount)
```

Comment:
We can add a new SYSCALL `CreateCallbackFromContractMethodName`, which creates the callback by hash and method name.

Reply:
Maybe we shouldn't provide this. Because it is not safe for the contracts that work with `invocationCounter`.